### PR TITLE
feat(assertion): update python example, assertion entity doc

### PR DESCRIPTION
**Pull request by mayurinehate**
This PR
- updates python example for emitting assertion info and its results
- updates document for assertion entity
- enables lint for metadata-ingestion/examples
- fixes graphql api to not fail if assertion platform is not set

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/assertion/AssertionMapper.java [hunk 1]
```diff
@@ -42,8 +42,9 @@ public static Assertion map(final EntityResponse entityResponse) {
     if (envelopedPlatformInstance != null) {
       result.setPlatform(mapPlatform(new DataPlatformInstance(envelopedPlatformInstance.getValue().data())));
     } else {
-      // Containers must have DPI to be rendered.
-      return null;
+      final DataPlatform unknownPlatform = new DataPlatform();
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/assertion/AssertionMapper.java:45]: Thank you thank you

File: metadata-models/docs/entities/assertion.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Assertion
+
+Assertion entity represents a data quality rule applied on dataset.  
+In future, it can evolve to span across Datasets, Flows (Pipelines), Models, Features etc.
+
+## Identity
+
+Assertion is identified by globally unique identifier, which is function of details that uniquely identify the assertion in an assertion platform.
```
Comment (jjoyce0510) [inline metadata-models/docs/entities/assertion.md:8]: Consider adding a sentence : For example, we compute the globally unique id for Great Expectations by ... 

Also consider adding a point about how the key is expected to be constant / stable across multiple executions of the same assertion

File: metadata-models/docs/entities/assertion.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Important Capabilities
```
Comment (jjoyce0510) [inline metadata-models/docs/entities/assertion.md:10]: Can we name this section as "Aspects"?

**Review by jjoyce0510 — COMMENTED** [review]
Overall this looks great to me. A few minor comments. Thank you!

File: metadata-models/docs/entities/assertion.md (hunk 3 above)
Comment (shirshanka) [reply metadata-models/docs/entities/assertion.md:10]: Let's be consistent across the board for these sections...  Datasets has `Important Capabilities` ... maybe we should call them `Features` or something else.. the goal was to separate this from the lower-level Aspects section which is auto-generated.

File: metadata-models/docs/entities/assertion.md (hunk 2 above)
Comment (jjoyce0510) [reply metadata-models/docs/entities/assertion.md]: I made this change

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**